pkg/util/test/util.go

```diff
@@ -412,6 +414,17 @@ func ValidateStorageCluster(
 		return err
 	}
 
+	// If portworx proxy is enabled, validate the portworx-proxy DaemonSet
+	if isPortworxProxyEnabled(liveCluster) {
```

Comment:
- We can move this to validateComponents just like the other components. 
- Call it ValidatePortworxProxy and validate both it is present and absent if enabled or disabled respectively. 
- Check for portworx-service, clusterrolebinding, service account, etc.

Look at stork, autopilot, csi validations for reference. Let's keep it consistent.